### PR TITLE
Ignore null or undefined values on insert, create and update schema

### PR DESCRIPTION
**Pull request by danderson00**

### CODE REVIEW

**Comment by danderson00** [comment]
Resolves https://github.com/Azure/azure-mobile-apps-node/issues/466

**Comment by danderson00** [comment]
@mamaso A quick once over would be great, thanks Cobber!

File: src/data/mssql/statements/insert.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ module.exports = function (table, item) {
 
     Object.keys(item).forEach(function (prop) {
         // ignore the property if it is an autoIncrement id
-        if ((prop !== 'id' || !table.autoIncrement) && item[prop] !== undefined) {
+        if ((prop !== 'id' || !table.autoIncrement) && item[prop] !== undefined && item[prop] !== null) {
```
Comment (mamaso) [inline src/data/mssql/statements/insert.js]: just looked it up, could consider the shorthand.  Then again, maybe not, as I needed to look it up..

```js
item == null  // is equivalent to
item === undefined && item === null
```

**Comment by mamaso** [comment]
lgtm, decide as you will on ```asdf==null``` shorthand

File: src/data/mssql/statements/insert.js (hunk 1 above)
Comment (danderson00) [inline src/data/mssql/statements/insert.js]: Not quite correct - an empty string and 0 are also falsy:

```Javascript
item == null // is equivalent to
item === null || item === undefined || item === 0 || item === ''
```

**Comment by danderson00** [comment]
Cheers mate!

File: src/data/mssql/statements/insert.js (hunk 1 above)
Comment (mamaso) [inline src/data/mssql/statements/insert.js]: They're falsey, but the above is incorrect.  Just tested, and

```js
null == null //true
null == undefined //true
null == 0 //false
null == '' //false
```

Comment (danderson00) [inline src/data/mssql/statements/insert.js]: Huh... This wasn't my understanding, and it makes no sense, but regardless, using obscure language features to slightly shorten code is never a good thing

Comment (mamaso) [inline src/data/mssql/statements/insert.js]: Agree, better to keep consistent with ```===``` usage anyway.  Interesting that there's a hacky comparison op for this usecase anyway

**Pull request merged**